microsoft-edge/webview2/concepts/working-with-local-content.md

```diff
@@ -0,0 +1,627 @@
+author: MSEdgeTeam
+ms.author: msedgedevrel
+ms.topic: conceptual
+ms.prod: microsoft-edge
+ms.technology: webview
+ms.date: 01/10/2023
+---
+# Working with local content in WebView2 apps
+
+In addition to loading remote content, content can also be loaded locally into WebView2.  There are several different approaches that can be used to load local content into a WebView2 control, including: 
+* Navigating to a file URL.
+* Navigating to an HTML string.
+* Virtual host name mapping.
+* Handling the `WebResourceRequested` event.
+
+These approaches are described below.
+
+
+<!-- ====================================================================== -->
+## Navigate to a file URL
+
+WebView2 allows navigations to file URLs, to load basic HTML or a PDF.  This is the most efficient and simples approach that satisfies the basic scenario.  However, it is less flexible than the other approaches.  This is the simplest approach to load local content.  However, like the browser, file URLs are limited in some capabilities. The document origin will be `null` for a file URL.  For each resource, the full path is needed to be specified.
+
+File URLs behave like they do in the browser.  For example, you can't make an `XMLHttpRequest` (XHR) in a file URL, because you're not working in the context of a webpage.  If the WebView2 control is loading local content, still came from a remote source (such as Bing.com), versus a file that came from the local drive.
+
+
+<!-- ------------------------------ -->
+#### Offline (local) vs. online (remote) content
+
+You must specify the full path of the file, for every resource.  For example:
+
+```
+C:\Users\username\Documents\GitHub\Demos\demo-to-do\index.html
+```
+
+For example, if you switch between online (remote) content, such as Bing.com and offline (local) content, you must specify the virtual domain.
+
+In contrast, if you navigate to a file URL, you have to replace it with the full path each time you switch between remote and local content.  When specifying a file URL, the app navigates to a specific path; that's the only way the app can load local content.
+
+You can't use cross-origin resources for navigating to a file URL; `document.origin` is null.
+
+You can specify a folder each time you switch between online (remote) and offline (local) content.
+
+##### [.NET/C#](#tab/dotnetcsharp)
+
+* [CoreWebView2.Navigate Method](/dotnet/api/microsoft.web.webview2.core.corewebview2.navigate)
+
+##### [WinRT/C#](#tab/winrtcsharp)
+
+* [CoreWebView2.Navigate Method](/microsoft-edge/webview2/reference/winrt/microsoft_web_webview2_core/corewebview2#navigate)
+
+##### [Win32/C++](#tab/win32cpp)
+
+* [ICoreWebView2::Navigate method](/microsoft-edge/webview2/reference/win32/icorewebview2#navigate)
+
+---
+
+
+<!-- ------------------------------ -->
+#### Example of a file URL
```

Comment:
@yildirimcagri can you provide more context on what this example is showing?

Reply:
The intended doc design here is:

h2: Doing x with a file url
h4: Here's what we mean by "a file url"
h4: Example of how to code a file url

Snapshot of page outline:
```
# Working with local content in WebView2 apps

## Navigate to a file URL
#### Offline (local) vs. online (remote) content
#### Example of a file URL
#### Example of navigating to a file URL

## Navigate to an HTML string
#### Example string representation of a webpage
#### Example of navigating to an HTML string

## Virtual host name mapping
#### Example of virtual host name mapping

## Handling the WebResourceRequested event
#### Example of handling the WebResourceRequested event

## See also
```